packages/@aws-cdk/aws-ecs/lib/cluster.ts

```diff
@@ -300,14 +300,24 @@ export class Cluster extends Resource implements ICluster {
       ? options.type
       : cloudmap.NamespaceType.DNS_PRIVATE;
 
-    const sdNamespace = namespaceType === cloudmap.NamespaceType.DNS_PRIVATE ?
-      new cloudmap.PrivateDnsNamespace(this, 'DefaultServiceDiscoveryNamespace', {
-        name: options.name,
-        vpc: this.vpc,
-      }) :
-      new cloudmap.PublicDnsNamespace(this, 'DefaultServiceDiscoveryNamespace', {
-        name: options.name,
-      });
+    let sdNamespace;
+    switch (namespaceType) {
+      case cloudmap.NamespaceType.DNS_PRIVATE:
+        sdNamespace = new cloudmap.PrivateDnsNamespace(this, 'DefaultServiceDiscoveryNamespace', {
+          name: options.name,
+          vpc: this.vpc,
+        });
+        break;
+      case cloudmap.NamespaceType.DNS_PUBLIC:
+        sdNamespace = new cloudmap.PublicDnsNamespace(this, 'DefaultServiceDiscoveryNamespace', {
+          name: options.name,
+        });
+        break;
+      case cloudmap.NamespaceType.HTTP:
+        sdNamespace = new cloudmap.HttpNamespace(this, 'DefaultServiceDiscoveryNamespace', {
+          name: options.name,
+        });
+    }
```

Comment:
What happens if we get to the end of this and `sdNamespace` is undefined? I know that there's not currently another type of namespace but let's add a default block which throws an error.